include/recognition/planar/HopcroftTarjan.hpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * IPlanarGraphRecognition.hpp
+ *
+ *  Created on: 24.03.2024
+ *      Author: Dzianis Lahunou
+ *      Ported by: Krzysztof Turowski (krzysztof.szymon.turowski@gmail.com)
+ */
+
+#pragma once
+
+#include "recognition/planar/IPlanarGraphRecognition.hpp"
+
+namespace Koala {
+
+const int left = 1;
+const int right = 2;
+/**
+ * @ingroup recognition
+ * The Interface for class for recognition of planar graphs p.
+ *
+ */
+
+class Block {
+   private:
+      std::list<int> Latt, Ratt;
+      std::list<NetworKit::WeightedEdge> Lseg, Rseg;
+   public:
+      Block(NetworKit::WeightedEdge e, std::list<int>& A){
+         Lseg.push_back(e);
+         Latt.splice(Latt.end(), A);
+      }
+      ~Block(){}
+      void flip() {
+         std::list<int> ha;
+         std::list<NetworKit::WeightedEdge> he;
+         ha.splice(ha.end(), Ratt); Ratt.splice(Ratt.end(), Latt); Latt.splice(Latt.end(), ha);
+         he.splice(he.end(), Rseg); Rseg.splice(Rseg.end(), Lseg); Lseg.splice(Lseg.end(), he);
+      }
+      int head_of_Latt() {
+         return Latt.front();
+      }
+      bool empty_Latt() {
+         return Latt.empty();
+      }
+      int head_of_Ratt() {
+         return Ratt.front();
+      }
+      bool empty_Ratt() {
+         return Ratt.empty();
+      }
```

Comment:
I guess since you use these functions only locally, you can remove them, they save next to nothing.